### ISSUE TITLE
Features for Fewest Moves - NISS

### ISSUE THREAD

**Issue by andersonaap** [larspetrus/Roofpig#12]
Would be wonderful can use roofpig
to visualize FMC reconstruction.

Enable a kind of token - like (switch) or (inverse) and (normal) 
that would would show a cube after applying a NISS (switch).

scramble: D2 U2 L' R2 B2 F2 U2 B U2 L D' F2 U B U B' D
solution
L2 F2 // 2x2x2
U' B U2 // pseudo 2x2x3
(switch)
U2 B U // 2x2x3 + EO
(switch)
B' R' B2 // Pseudo F2L
(switch)
B' L' F D2 F' L B2 // AB3C

We could visualize this kind of reconstruction in.

L2 F2 U' B U2 (switch) U2 B U (switch) B' R' B2 (switch) B' L' F D2 F' L B2

**Comment by larspetrus** [comment]
This is really interesting. Visualizing NISS solutions is an unsolved problem, AFAIK, and makes the sport very hard to follow.

Let me look into some ideas for a few days!

**Comment by larspetrus** [comment]
This isn't quite what you asked for, but I made a NISS solution visualizing web page that I hope can be a good start.

It's at http://birdflu.lar5.com/fmc and you can enter a NISS format solution and an (optional) scramble, and get an animation of each step.

A few premade solutions I grabbed from http://fmcsolves.cubing.net/?tag=niss:

http://birdflu.lar5.com/fmc?utf8=%E2%9C%93&scramble=R2+F+D2+F%27+R2+B%27+D2+B%27+F%27+L2+F+U%27+F2+L%27+U2+B+R%27+F+D2+F+D%27&niss=L2+D%27+L2+%28R2+F2%29+R+B+R%27+%28R2+F+R%29+R+L+F%27+L%27+F+R%27+U2+F%27+U+F+U2+F+U+F%27+U%27+R%27+F%27+L+F+L%27+R&commit=Show

http://birdflu.lar5.com/fmc?utf8=%E2%9C%93&scramble=R2+F+D2+F%27+R2+B%27+D2+B%27+F%27+L2+F+U%27+F2+L%27+U2+B+R%27+F+D2+F+D%27&niss=L2+D%27+L2+%28R2+F2%29+R+B+R%27+%28R2+F+R%29+R+L+F%27+L%27+F+R%27+U2+F%27+U+F+U2+F+U+F%27+U%27+R%27+F%27+L+F+L%27+R&commit=Show

Your example (missing an insertion):
http://birdflu.lar5.com/fmc?utf8=%E2%9C%93&scramble=D2+U2+L%27+R2+B2+F2+U2+B+U2+L+D%27+F2+U+B+U+B%27+D&niss=L2+F2+U%27+B+U2+%28U2+B+U%29+B%27+R%27+B2+%28B%27+L%27+F+D2+F%27+L+B2%29&commit=Show

**Comment by andersonaap** [comment]
That really nice good start. Great job.

**Comment by andersonaap** [comment]
Do you intend to add the Nissualizer page in roofpig repository or create another one?

**Comment by larspetrus** [comment]
What's missing for it to be useful?

I'm working on a database of all LL algs (http://birdflu.lar5.com/), and that code isn't public yet. I put the Nissualizer there, since it's easy.

**Comment by andersonaap** [comment]
@larspetrus 
Sugestions:
- put solution in a textarea. each line will be a step displayed in roofpig.
- let white on top, green on fron or offer a option for choosing orientation

what about ?
https://run.plnkr.co/7eQroFCZDDh24RNn/

maybe in showing normal/inversed pair of cubes before which step. 
or just a one cube that step was done.

**Comment by andersonaap** [comment]
http://codepen.io/anon/pen/LkyEkW?editors=1000
